### ISSUE TITLE
Add file tree view like Sublime Text

### ISSUE THREAD

**Issue by liujianping** [MacDownApp/macdown#141]
add file tree view just like sublime text will be more helpful...:)

**Comment by daveqr** [comment]
Agree 100% with @nickcheng. I also have a lot of md files in a project, and it's much more difficult to work with them as it is now.

**Comment by hayd** [comment]
When you are writing a book, you have multiple chapters. If you do this in a single markdown file after a chapter or so the rendering panel is usually completely out-of-sync with the text. Even if all the file tree did was switch between windows that would be really useful. ...though I don't really understand the problem of keeping multiple buffers open (many text editors do it).

I wouldn't want it open all the time, but being able to toggle the tree would be excellent. (I love NERDTree in vim!)

**Comment by DEfusion** [comment]
+1 for an optional sidebar file tree

**Comment by timotheecour** [comment]
Tabs are not the most important (hard to read tab titles if too many tabs, because of horizontal layout), but something like sublimetext's Side Bar would be perfect. 
Showing tree view would be a nice extra but the most important feature would be showing the set of opened files as a side bar on the left hand side, stacked vertically (exactly as sublime text)

**Comment by uranusjr** [comment]
Tree views without tabs would be very awkward. What should the app do when you open a document from the tree?

@orta built a pretty interesting solution on this issue [a while ago](https://twitter.com/orta/status/584850525910343680). Haven’t heard much since then though. It would be brilliant if someone could work on this popup idea; I would definitely merge it if it’s done reasonably.

**Comment by peacedove** [comment]
Tree view without tabs is like what [NERDTree](https://youtu.be/CPu9mDpSYj0?t=1m58s) is doing. It is not awkward at all and is actually very helpful.

As @nickcheng had mentioned, [Ulysses](http://www.ulyssesapp.com/) also uses a similar interface.

**Comment by carinelaine** [comment]
+1 for tree view / file list. Tabs are not necessary, it could work just like Ulysses, or Evernote. Maybe with an option to either auto save the file, or popup for confirmation.

I mostly use MacDown for documentation, and I have quite a few .md files in a project, going back and forth between them as I work.

**Comment by benhaynes** [comment]
Another request for this optional sidebar feature. For those editing many files, it would be invaluable.

**Comment by aalasolutions** [comment]
+1 and hoping it comes soon

**Comment by masukomi** [comment]
I agree with @uranusjr that @orta 's [idea of a quick find](https://twitter.com/orta/status/584850525910343680) would solve the problem nicely. It addresses the switching between files that people want and it doesn't use up screen real estate.

**Comment by debradley** [comment]
@masukomi objects that this feature isn't useful to him, but his use cases aren't everyone's - he's a writer but I'm a developer. We likely use MacDown in very different ways.

I work with many small, nested files and switching between them is incredibly inconvenient. As a web developer I work with many different technologies and keeping them all straight is difficult, so I write cheat sheets to help me. Here's an example of the structure I'm using:

```
Notes
  JavaScript
    Node
      index.md
      install.md
      configure.md
      update.md
      ...
    Yarn
      index.md
      install.md
      configure.md
      update.md
      ...
    ...
  Ruby
    index.md
    install.md
    ...
      Rails
        index.md
        install.md
          ...
          ActiveRecord
            ...
    ...
  ...
```

As you can imagine from this example, because I rely on multiple folder levels and because many of my files share names, a tree view would be immensely useful and a quick find feature, while helpful, would be of limited use.

Nor do I see think screen real estate has to be an issue. The tree view should be opt-in, not showing up unless you specifically choose to open a folder. Further, you should be able to toggle the visibility of the tree view on and off even once a folder has been opened.

**Comment by uranusjr** [comment]
@debradley Just want to make sure I understand you correctly… Who do you mean by “him?” Orta most definitely is a developer, and I think I qualify as one as well. And I am most certainly not a writer.

**Comment by debradley** [comment]
"Him" in this case is @masukomi - which is why I tagged him. I was responding this his comments in this thread, including [this one](https://github.com/MacDownApp/macdown/issues/141#issuecomment-55063280) where he discussed his needs as a writer. My point is that that there are multiple use cases for the app and that they can coexist.

**Comment by yjlintw** [comment]
@debradley You can try vscode with extensions. [Markdown + Math](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath) is the one that I am using right now.

**Comment by carlowahlstedt** [comment]
Honestly, I agree with @debradley. It could be added in such a way that a user who never previously needed or used it would notice. From a UX standpoint, a simple menu option to enable the view of it.

**Comment by yjlintw** [comment]
I'm not sure why I get a thumbs down from @carlowahlstedt. VSCode + extensions is a pretty good alternative. Apparently the author is not interested in implementing this feature to MacDown, since this issue is already open three years. If VSCode + extension does not fit your need, you can elaborate that, and maybe I can find another better alternatives.

**Comment by carlowahlstedt** [comment]
That's funny @yjlintw I was just thinking that I should probably clarify and looking at the repo to see if I could try to implement this is some un-obtrusive way. However, my objective-c is rough and I'm not sure I'd be successful. That said, I agree is a pretty good alternative, but still not better than Macdown actually having this as feature. I keep VSCode open almost all day just to do this feature and then open more windows when doing various development tasks. It can make it annoying when having multiple windows open and wanting to just go to the markdown editor or the dev tools.

**Comment by lovelock** [comment]
> Agree 100% with @nickcheng. I also have a lot of md files in a project, and it's much more difficult to work with them as it is now.

This is exactly why Typora is so popular now

**Comment by fosple** [comment]
> Honestly, I agree with @debradley. It could be added in such a way that a user who never previously needed or used it would notice. From a UX standpoint, a simple menu option to enable the view of it.

+1 

I use MacDown mainly as a replacement for accessing notes on my computer (instead of Apple Notes, Bear,...). A optional tree like navigation structure would really benefit the UX and not hurt users who do not want it.

Recently I discovered https://github.com/lukakerr/Pine/ which does exactly this and it feels really good from a UX perspective. However, currently it still is buggy and I really love the stability of MacDown a lot. So I would love to stick with MacDown :)

**Comment by felicitia** [comment]
> > Agree 100% with @nickcheng. I also have a lot of md files in a project, and it's much more difficult to work with them as it is now.
> 
> This is exactly why Typora is so popular now

Thank you for the recommendation. I just downloaded macdown today and this is the first thing I noticed missing since I always have multiple files in my projects (e.g., website). So I just tried Typora and it does exactly what I wanted. It has tree view of the file structure and also an outline view. People may have different needs though. I've always been using Sublime and I really depend on the tree view, but it doesn't preview .md very well.

**Comment by ThePredators** [comment]
That would be great to open folders in MacDown.
